### PR TITLE
fix: Remove 3-dots menu actions role redactor filter - EXO-63325

### PR DESCRIPTION
**Pull request by hakermi**
Prior to this change, 3-dots menu actions are filtered by a redactor role inside a redactor space. This PR removes the redactor role actions filter and enhance the extensions filtering.

### CODE REVIEW

File: documents-webapp/src/main/webapp/vue-app/documents/components/body/table/DocumentActionMenu.vue [hunk 1]
```diff
@@ -29,23 +29,15 @@ export default {
     menuExtensionApp: 'DocumentMenu',
     menuExtensionType: 'menuActionMenu',
     menuExtensions: {},
-    mobileOnlyExtensions: ['favorite','details'],
-    desktopOnlyExtensions: ['edit'],
     editExtensions: ['edit'],
-    fileOnlyExtension: ['download','favorite'],
-    redactorExtension: ['download', 'versionHistory','details', 'copyLink'],
```
Comment (rdenarie) [inline documents-webapp/src/main/webapp/vue-app/documents/components/body/table/DocumentActionMenu.vue:36]: why did you remove mobileOnly, destopOnly and fileOnly filters ?

We only wnat to remove redatorExtension

File: documents-webapp/src/main/webapp/vue-app/documents/components/body/table/DocumentActionMenu.vue [hunk 2]
```diff
@@ -29,23 +29,15 @@ export default {
...
   }),
   created() {
     document.addEventListener(`extension-${this.menuExtensionApp}-${this.menuExtensionType}-updated`, this.refreshMenuExtensions);
     document.addEventListener('documents-supported-document-types-updated', this.refreshSupportedDocumentExtensions);
-    this.$transferRulesService.getDocumentsTransferRules().then(rules => {
-      this.sharedDocumentSuspended = rules.sharedDocumentStatus === 'true';
-      this.downloadDocumentSuspended = rules.downloadDocumentStatus === 'true';
-      this.refreshMenuExtensions();
-    });
+    this.refreshMenuExtensions();
```
Comment (rdenarie) [inline documents-webapp/src/main/webapp/vue-app/documents/components/body/table/DocumentActionMenu.vue:40]: Whydo you remove this part ?

File: documents-webapp/src/main/webapp/vue-app/documents/components/body/table/DocumentActionMenu.vue (hunk 1 above)
Comment (hakermi) [reply documents-webapp/src/main/webapp/vue-app/documents/components/body/table/DocumentActionMenu.vue:36]: I moved the check to the enabled function of the extension so we will not loop in extensions on each filter. It is an improvement

File: documents-webapp/src/main/webapp/vue-app/documents/components/body/table/DocumentActionMenu.vue (hunk 2 above)
Comment (hakermi) [reply documents-webapp/src/main/webapp/vue-app/documents/components/body/table/DocumentActionMenu.vue:40]: Moved to the extension

**Review by rdenarie — APPROVED** [review]
